jplag/src/main/java/de/jplag/GreedyStringTiling.java

```diff
@@ -217,18 +177,23 @@ private void addMatchIfNotOverlapping(List<Match> matches, int startA, int start
         matches.add(new Match(startA, startB, length));
     }
 
-    /**
-     * Disable finding a match at separator tokens and basecode matches for non-basecode comparisons.
-     * @param tokenList Tokens to mark.
-     * @param isBaseCodeComparison Whether the {@link Token#basecode} matches should be enabled for matching.
-     */
-    private void markTokens(TokenList tokenList, boolean isBaseCodeComparison) {
-        for (Token token : tokenList.allTokens()) {
-            if (isBaseCodeComparison) {
-                token.setMarked(token.type == FILE_END || token.type == SEPARATOR_TOKEN);
-            } else {
-                token.setMarked(token.type == FILE_END || token.type == SEPARATOR_TOKEN || (token.isBasecode() && options.hasBaseCode()));
+    private Set<Token> initiallyMarkedTokens(List<Token> tokens, boolean isBaseCodeComparison) {
+        Set<Token> markedTokens = new HashSet<Token>();
```

Comment:
Consider an identity HashSet (otherwise we rely on equals and hashcode of the token, not sure whether we want this.

`Collections.newSetFromMap(new IdentityHashMap<>())` .. but not sure here

Reply:
I am aiming to refactor `Token` to a record class thus `equals` and `hashCode` should get generated correctly.
However, it might still makes sense to use an identity HashSet as reference equality suffices for our use case. Do you know how the performance between HashSet and IdentityHashSet compares?

Reply:
HashSet uses equals, IdentityHashMap (so also the set) uses ==

Reply:
Will be overhauled in #623 anyways where we can't use identity anymore. Thus is it fine if we keep the `HashSet` here? We can later still do performance optimisation.